test/lib/ufe/testConnections.py

```diff
@@ -421,5 +423,99 @@ def testConnectionsHandler(self):
         conns = connections.allConnections()
         self.assertEqual(len(conns), 1)
 
+    def testCreateStandardSurface(self):
+        '''Test create a working standard surface shader.'''
+        #
+        #
+        # We start with standard code from testContextOps:
+        #
+        #
+        cmds.file(new=True, force=True)
+
+        # Create a proxy shape with empty stage to start with.
+        import mayaUsd_createStageWithNewLayer
+        proxyShape = mayaUsd_createStageWithNewLayer.createStageWithNewLayer()
+
+        # Create a ContextOps interface for the proxy shape.
+        proxyPathSegment = mayaUtils.createUfePathSegment(proxyShape)
+        proxyShapePath = ufe.Path([proxyPathSegment])
+        proxyShapeItem = ufe.Hierarchy.createItem(proxyShapePath)
+        contextOps = ufe.ContextOps.contextOps(proxyShapeItem)
+
+        cmd = contextOps.doOpCmd(['Add New Prim', 'Capsule'])
+        ufeCmd.execute(cmd)
+        cmd = contextOps.doOpCmd(['Add New Prim', 'Material'])
+        ufeCmd.execute(cmd)
+
+        rootHier = ufe.Hierarchy.hierarchy(proxyShapeItem)
+
+        materialItem = rootHier.children()[-1]
+        materialAttrs = ufe.Attributes.attributes(materialItem)
+        contextOps = ufe.ContextOps.contextOps(materialItem)
+
+        cmd = contextOps.doOpCmd(['Add New Prim', 'Shader'])
+        ufeCmd.execute(cmd)
+
+        materialHier = ufe.Hierarchy.hierarchy(materialItem)
+        materialPrim = usdUtils.getPrimFromSceneItem(materialItem)
+
+
+        shaderItem = materialHier.children()[0]
+        shaderAttrs = ufe.Attributes.attributes(shaderItem)
+        shaderPrim = usdUtils.getPrimFromSceneItem(shaderItem)
+
+        shaderAttr = shaderAttrs.attribute("info:id")
+        shaderAttr.set("ND_standard_surface_surfaceshader")
+        #
+        #
+        # Then switch to connection code to connect the shader. Since we never created the
+        # attributes, we expect the connection code to create them.
+        #
+        #
+        shaderOutput = shaderAttrs.attribute("outputs:out")
+        materialOutput = materialAttrs.attribute("outputs:surface")
+
+        connectionHandler = ufe.RunTimeMgr.instance().connectionHandler(materialItem.runTimeId())
+
+        # The attributes are not created yet:
+        self.assertEqual(len(materialPrim.GetAuthoredProperties()), 0)
+        self.assertEqual(len(shaderPrim.GetAuthoredProperties()), 1)
+        self.assertIn("info:id", [i.GetName() for i in shaderPrim.GetAuthoredProperties()])
```

Comment:
As the code previously checks that there is only one entry, this line can directly access the first (and only) entry i.e. avoid the loop.